hazelcast-jet-core/src/main/java/com/hazelcast/jet/datamodel/KeyedWindowResult.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.datamodel;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+
+import static com.hazelcast.jet.impl.util.Util.toLocalTime;
+
+/**
+ * Holds the result of a group-and-aggregate operation performed over a
+ * time window.
+ *
+ * @param <K> type of the grouping key
+ * @param <R> type of the aggregated result
+ */
+public class KeyedWindowResult<K, R> extends WindowResult<R> {
```

Comment:
I think it would be better to implement Map.Entry here, or extend `TimestampedEntry` so that it can be directly written to map sink. WindowResult can be perhaps an interface?